### PR TITLE
feat: Add the reason for map camera movement

### PR DESCRIPTION
**Pull request by barbeau**
As discussed in https://github.com/googlemaps/android-maps-compose/issues/49, this PR exposes the `reason` value passed back in `GoogleMap.OnCameraMoveStartedListener(int reason)` callback so that developers can tell the reason that the camera started moving (e.g., user gesture, developer animation, API animation):
https://developers.google.com/android/reference/com/google/android/gms/maps/GoogleMap.OnCameraMoveStartedListener#onCameraMoveStarted(int)

For constant values see:
https://developers.google.com/android/reference/com/google/android/gms/maps/GoogleMap.OnCameraMoveStartedListener#constants

The PR also adds demo code to `LocationTrackingActivity` to print the reason that the camera moves, as well as assertions to test behavior in an existing unit test.

---

Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open a GitHub issue as a bug/feature request before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes https://github.com/googlemaps/android-maps-compose/issues/49 🦕

### CODE REVIEW

**Review by arriolac — COMMENTED** [review]
Change generally LGTM although I think this is a good opportunity to improve the type safety around `reason` and use an enum instead. A new type can also be created for the initial case. Note that this is a pattern already used in the library for `MapType`.

**Comment by barbeau** [comment]
@arriolac Good idea! I convert to an enum in https://github.com/googlemaps/android-maps-compose/pull/154/commits/819179ec6d98e27b54740a8b6c0eefa5ae4dcee3.

I wasn't sure how to handle the case where a new Maps SDK constant is passed back from the `GoogleMap.OnCameraMoveStartedListener` that isn't known to the maps-compose library (e.g., if the Maps SDK is updated). I ended up mapping these unknown constants to another `CameraMoveStartedReason` enum value ` UNKNOWN(-2)`, but let me know if you have a better idea of how to handle this.

File: maps-compose/src/main/java/com/google/maps/android/compose/CameraPositionState.kt [hunk 1]
```diff
@@ -27,12 +27,7 @@ import com.google.android.gms.maps.GoogleMap
 import com.google.android.gms.maps.Projection
 import com.google.android.gms.maps.model.CameraPosition
 import com.google.android.gms.maps.model.LatLng
-import kotlinx.coroutines.CancellableContinuation
-import kotlinx.coroutines.CancellationException
-import kotlinx.coroutines.Job
-import kotlinx.coroutines.cancel
-import kotlinx.coroutines.currentCoroutineContext
-import kotlinx.coroutines.suspendCancellableCoroutine
+import kotlinx.coroutines.*
```
Comment (arriolac) [inline maps-compose/src/main/java/com/google/maps/android/compose/CameraPositionState.kt:30]: Nit: Use single name imports here. I have my preferences configured to the following:

image

This should probably be enforced in the linter.

Comment (barbeau) [reply maps-compose/src/main/java/com/google/maps/android/compose/CameraPositionState.kt]: Ah, good catch - I fixed the imports to not use wildcards and updated my AS settings.

**Pull request merged**